### PR TITLE
Laboratory work #1, Zakharova Sonia - 21FPL2

### PR DESCRIPTION
**Pull request by solonarr**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -11,6 +11,17 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:14]: 👍🏻

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,6 +11,17 @@ def tokenize(text: str) -> list[str] | None:
...
+        return None
+    tokens = ''
+    punc = ''''1234567890!"#$%&'()’*º+,-./:;<=>?@[\\]^_`{|}~'''
+    text = text.replace(' ', '')
+    text = text.replace('\n', '')
+    for token in text:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:20]: Use list comprehension

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -19,6 +30,13 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+    my_dict = {token: tokens.count(token)/len(tokens) for token in tokens}
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:38]: add spaces aroun `/`

File: lab_1_classify_by_unigrams/target_score.txt [hunk 4]
```diff
@@ -1 +1 @@
-0
+6
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/target_score.txt:1]: what about 8? 10?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
One of the best implementation, please continue

**Comment by solonarr** [comment]
arrrr code style

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py]: done

**Comment by solonarr** [comment]
i do not understand what mypy wants from me...

**Comment by solonarr** [comment]
one last error.....

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -11,6 +11,13 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    punc = ''''1234567890!"#$%&'()’*º+,-./:;<=>?@[\\]^_`{|}~'''
+    text = text.replace(' ', '')
+    text = text.replace('\n', '')
+    tokens = [token.lower() for token in text if token not in punc]
+    return [*tokens]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:20]: why do you do list from list ?

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -19,6 +26,13 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+    my_dict = {token: tokens.count(token) / len(tokens) for token in tokens}
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:34]: do not create a function, return directly

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -28,6 +42,13 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not (isinstance(language, str) and isinstance(text, str)):
+        return None
+    text_to_proceed = tokenize(text)
+    prof_dict = calculate_frequencies(text_to_proceed)
+    if prof_dict:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:49]: if not -> return None

File: lab_1_classify_by_unigrams/target_score.txt [hunk 8]
```diff
@@ -1 +1 @@
-0
+8
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/target_score.txt:1]: let's go to 10!

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:20]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:34]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -11,6 +12,12 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    punc = ''''1234567890!"#$%&'()’*º+,-./:;<=>?@[\\]^_`{|}~'''
+    text = text.replace(' ', '')
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: in one line

Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:18]: Why do not you add this symbols to punctuation?

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -63,6 +114,20 @@ def detect_language(
...
+        return None
+    unknown_and_1 = compare_profiles(unknown_profile, profile_1)
+    unknown_and_2 = compare_profiles(unknown_profile, profile_2)
+    if (isinstance(unknown_and_1, float)
+            and isinstance(unknown_and_2, float)):
+        if unknown_and_1 < unknown_and_2:
+            return str(profile_1['name'])
+        if unknown_and_2 < unknown_and_1:
+            return str(profile_2['name'])
+    detected = sorted([str(profile_1.get('name')), str(profile_2.get('name'))])[0]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:129]: do not create the variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Very good

File: lab_1_classify_by_unigrams/main.py (hunk 10 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:129]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 9 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:18]: fixed???????

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -19,6 +24,12 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:29]: two loops, please mer it

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -80,6 +149,18 @@ def preprocess_profile(profile: dict) -> dict[str, str | dict] | None:
...
     with relative frequencies without unnecessary ngrams
     """
+    if not (isinstance(profile, dict)
+            and 'name' in profile
+            and 'freq' in profile
+            and 'n_words' in profile):
+        return None
+    freq_dict = {}
+    for unigram in profile['freq']:
+        if unigram.lower() in freq_dict:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:159]: `unigram.lower()`,  `profile['n_words'][0]`, `profile['freq'][unigram]` to variables, please

File: lab_1_classify_by_unigrams/main.py [hunk 13]
```diff
@@ -88,6 +169,20 @@ def collect_profiles(paths_to_profiles: list) -> list[dict[str, str | dict[str,
...
     :return: a list of loaded profiles
     """
+    if not isinstance(paths_to_profiles, list):
+        return None
+    collect_profs = []
+    for prof in paths_to_profiles:
+        if not load_profile(prof):
+            return None
+        load_prof = load_profile(prof)
+        if load_prof:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:179]: Why do not in the same way as `load_profile` ?

File: lab_1_classify_by_unigrams/main.py [hunk 14]
```diff
@@ -98,10 +193,19 @@ def detect_language_advanced(unknown_profile: dict[str, str | dict[str, float]],
...
+    det_lang = sorted(det_lang, key=lambda tpl: (tpl[1], tpl[0]))
+    return det_lang
 
 
 def print_report(detections: list[tuple[str, float]]) -> None:
     """
     Prints report for detection of language
     :param detections: a list with distances for each available language
     """
+    if isinstance(detections, list):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:209]: else? reverse the condition

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 11 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:29]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 12 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:159]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 13 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:179]: aaah

Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:179]: because mypy wanted me to be sure that the variable in preprocess_rpofile was really a dict

Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:179]: yaay fixed

File: lab_1_classify_by_unigrams/main.py (hunk 14 above)
Comment (solonarr) [reply lab_1_classify_by_unigrams/main.py:209]: fixed??????

**Pull request closed without merging**